Source/UnitEx.cpp

```diff
@@ -65,7 +65,17 @@ struct cmdMove // same for patrol, dock, dockEG, cargo route
 	// ...
 };
 
-// todo: cargoroute
+struct cmdCargoRoute
+{
+	char numUnits;
+	short unitId;
+	short numWayPoints;
+	long pts[3];
+	short mineWayPointIndex;
+	short smelterWayPointIndex;
+	short mineUnitId;
+	short smelterUnitId;
+};

```

Comment:
I'm not sure how I would do this, in practice.  How would I "combine" the two structs into the packet's data buffer?

Reply:
Basically with casting and pointer arithmetic. I suppose that could potentially be easier if the array wasn't part of the struct.

At any rate, if you're building the struct specifically for that one method, it may be easier to just use a single struct, and perhaps default the values that have fixed assumptions.

Might be cleaner if the scope of the struct could be limited to the function that the assumptions are tied to.